src/EditorFeatures/CSharpTest/AddStaticModifier/AddStaticModifierTests.cs

```diff
@@ -0,0 +1,111 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using Microsoft.CodeAnalysis.CSharp.AddStaticModifier;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Editor.CSharp.UnitTests.Diagnostics;
+using Microsoft.CodeAnalysis.Test.Utilities;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.Editor.CSharp.UnitTests.AddStaticModifier
+{
+    public class AddStaticModifierTests : AbstractCSharpDiagnosticProviderBasedUserDiagnosticTest
+    {
+        internal override (DiagnosticAnalyzer, CodeFixProvider) CreateDiagnosticProviderAndFixer(Workspace workspace)
+            => (null, new CSharpAddStaticModifierCodeFixProvider());
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsAddStaticModifier)]
+        public async Task TestField()
+        {
+            await TestInRegularAndScript1Async(
+@"
+public static class Foo
+{
+    int [||]i;
+}",
+@"
+public static class Foo
+{
+    static int i;
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsAddStaticModifier)]
+        public async Task TestMethod()
+        {
+            await TestInRegularAndScript1Async(
+@"
+public static class Foo
+{
+    void [||]M() { }
+}",
+@"
+public static class Foo
+{
+    static void M() { }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsAddStaticModifier)]
+        public async Task TestProperty()
+        {
+            await TestInRegularAndScript1Async(
+@"
+public static class Foo
+{
+    object [||]P { get; set; }
+}",
+@"
+public static class Foo
+{
+    static object P { get; set; }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsAddStaticModifier)]
+        public async Task TestEventField()
+        {
+            await TestInRegularAndScript1Async(
+@"
+public static class Foo
+{
+    event System.Action [||]E;
+}",
+@"
+public static class Foo
+{
+    static event System.Action E;
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsAddStaticModifier)]
+        public async Task FixAll()
+        {
+            await TestInRegularAndScript1Async(
+@"namespace NS
+{
+    public static class Foo
+    {
+        int {|FixAllInDocument:|}i;
+        void M() { }
+        object P { get; set; }
+        event System.Action E;
+    }
+}",
+@"namespace NS
+{
+    public static class Foo
+    {
+        static int i;
+
```

Comment:
An empty line was inserted by the code-fix. I am not sure whether that's what we want or not so I'll wait for you input.

Reply:
it's not great.  ideally this wouldnt' happen.  but i wouldn't block this feature on this issue.

Reply:
As I am not explicitly adding any trailing return, could it be because of the editor configuration?

Reply:
Agree with @CyrusNajmabadi

Reply:
@mavasani this point isn't fixed, shall I create a separate ticket?

Reply:
Yes, please file a separate tracking issue.